lib/scale/enc_dec_test.go

```diff
@@ -189,3 +189,19 @@ func TestEncodeDecode_Array(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, b, dec)
 }
+
+func TestEncodeDecodeSliceStruct(t *testing.T) {
+	type SimpleStruct struct {
+		A int64
+		B bool
+	}
```

Comment:
could you add some more cases, like structs with `[]byte` fields, `common.Hash` fields, maybe fields with custom decode functions also?

Reply:
Done